### PR TITLE
fix(dynamic-sampling): ensure that at least one project has count

### LINKED ISSUE #81556: ZeroDivisionError: float division by zero
Sentry Issue: [SENTRY-3J60](https://sentry.sentry.io/issues/6074803646/?referrer=github_integration)

```
ZeroDivisionError: float division by zero
  File "sentry/dynamic_sampling/models/common.py", line 28, in guarded_run
    return model.run(model_input)
  File "sentry/dynamic_sampling/models/base.py", line 38, in run
    return self._run(model_input)
  File "sentry/dynamic_sampling/models/projects_rebalancing.py", line 33, in _run
    result, _ = full_rebalancing.run(
  File "sentry/dynamic_sampling/models/base.py", line 38, in run
    return self._run(model_input)
  File "sentry/dynamic_sampling/models/full_rebalancing.py", line 74, in _run
    new_sample_rate = desired_count / count
```

### PR DESCRIPTION
**Pull request by constantinius**
Make sure that at least one project with counts is present before doing a Project Rebalancing run

Closes https://github.com/getsentry/sentry/issues/81556

### CODE REVIEW

File: tests/sentry/dynamic_sampling/tasks/test_boost_low_volume_projects.py [hunk 1]
```diff
@@ -183,6 +184,22 @@ def test_project_mode_sampling_with_query(self):
 
         assert get_guarded_project_sample_rate(org1, p1) == 0.2
 
+    @with_feature(["organizations:dynamic-sampling", "organizations:dynamic-sampling-custom"])
+    def test_project_mode_sampling_with_query_zero_metrics(self):
+        org1 = self.create_organization("test-org")
+        p1 = self.create_project(organization=org1)
```
Comment (shellmayr) [inline tests/sentry/dynamic_sampling/tasks/test_boost_low_volume_projects.py:190]: nit: can we use more obvious names like organization and project here?

File: tests/sentry/dynamic_sampling/tasks/test_boost_low_volume_projects.py [hunk 2]
```diff
@@ -183,6 +184,22 @@ def test_project_mode_sampling_with_query(self):
...
+
+        org1.update_option("sentry:sampling_mode", DynamicSamplingMode.PROJECT)
+        p1.update_option("sentry:target_sample_rate", 0.2)
+
+        # bulk task
```
[the remark this answers is not in the document]
Comment (constantinius) [reply tests/sentry/dynamic_sampling/tasks/test_boost_low_volume_projects.py:195]: Copy-paste error, thanks

File: src/sentry/dynamic_sampling/tasks/boost_low_volume_projects.py [hunk 3]
```diff
@@ -398,6 +398,10 @@ def adjust_sample_rates_of_projects(
         )
         return
 
+    # The rebalancing will not work (or would make sense) when we have only projects with zero-counts.
+    if not any(count > 0 for _, count, _, _ in projects_with_tx_count):
+        return
+
     projects_with_counts = {
         project_id: count_per_root for project_id, count_per_root, _, _ in projects_with_tx_count
     }
```
Comment (shellmayr) [inline src/sentry/dynamic_sampling/tasks/boost_low_volume_projects.py:407]: Could this be slightly more readable?

```suggestion
    projects_with_counts = {
        project_id: count_per_root for project_id, count_per_root, _, _ in projects_with_tx_count
    } 
    # The rebalancing will not work (or would make sense) when we have only projects with zero-counts.
    if sum(projects_with_counts.values()) == 0:
        return
```

**Review by shellmayr — APPROVED** [review]
LGTM

**Pull request merged**